gunicorn/arbiter.py

```diff
@@ -494,6 +496,11 @@ def manage_workers(self):
 
     def spawn_worker(self):
         self.worker_age += 1
+        worker_id = -1
+        for i in range(self.num_workers):
```

Comment:
why not then simply reusing worker.age which is unique and incremented once
a worker is created?
On Wed, 30 Sep 2015 at 18:41, Adam Weiner <notifications@github.com> wrote:

> In gunicorn/arbiter.py
> <https://github.com/benoitc/gunicorn/pull/1118#discussion_r40818084>:
>
> > @@ -494,6 +496,11 @@ def manage_workers(self):
> >
> >      def spawn_worker(self):
> >          self.worker_age += 1
> > +        worker_id = -1
> > +        for i in range(self.num_workers):
>
> Thanks for all the feedback, @benoitc <https://github.com/benoitc>.
>
> Despite the added overhead, I think one of the most useful parts of
> instance identification is having the instance number as a basic counter in
> a predictable range, as opposed to a hash or other unique identifier. If
> implementing this with a hash it wouldn't really add much to gunicorn, as
> each worker's PID can already act as such a unique identifier.
>
> The use cases for this are plentiful, but could include workers logging
> under the same set of names every time an application is run with gunciorn
> (app0, app1, etc), or having workers listen on the same set of incremental
> ports (1000, 1001, etc). I think the predictability that a basic counter
> provides is pretty important here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/benoitc/gunicorn/pull/1118/files#r40818084>.
>